plugin-java/src/main/java/appland/javaAgent/AppMapJavaAgentDownloadService.java

```diff
@@ -0,0 +1,173 @@
+            // e.g. on system where symbolic links are unsupported
+            Files.copy(assetDownloadPath, agentFilePath);
+        }
+
+        return true;
+    }
+
+    /**
+     * Guards the operation of the runnable with a lock file.
+     * If the lock file does already exist and is less than 5 minutes old, then the runnable is NOT executed.
+     * Otherwise, the lock file is either created or updated before the runnable is executed.
+     *
+     * @param lockFilePath Target path of the lock file
+     * @param runnable     The operation to guard with the lock file
+     * @return {@code true} if the runnable was executed
+     * @throws IOException Thrown if the runnable failed to execute
+     */
+    private boolean guardWithLockFile(@NotNull Path lockFilePath,
+                                      @NotNull ThrowableRunnable<IOException> runnable) throws IOException {
+        if (Files.exists(lockFilePath)) {
+            var fiveMinutesAgo = Instant.now().minus(5, ChronoUnit.MINUTES);
+            if (!Files.getLastModifiedTime(lockFilePath).toInstant().isBefore(fiveMinutesAgo)) {
+                return false;
+            }
+
+            Files.setLastModifiedTime(lockFilePath, FileTime.from(Instant.now()));
+        } else {
+            Files.createFile(lockFilePath);
+        }
+
+        try {
+            runnable.run();
+        } finally {
+            Files.deleteIfExists(lockFilePath);
+        }
+        return true;
+    }
+
+    /**
+     * Downloads the GitHub asset and stores it at the given path.
+     *
+     * @param indicator         Progress indicator
+     * @param agentReleaseAsset Asset to download
+     * @param assetDownloadPath Target path, where the download is stored on disk
+     * @throws IOException Thrown if the download failed to complete
+     */
+    private static void downloadAgentJarFile(@NotNull ProgressIndicator indicator,
+                                             @NotNull GitHubReleaseAsset agentReleaseAsset,
+                                             @NotNull Path assetDownloadPath) throws IOException {
+        try {
+            agentReleaseAsset.download(indicator, assetDownloadPath);
+        } catch (IOException e) {
+            // remove partial download and throw IOException
+            Files.deleteIfExists(assetDownloadPath);
+            throw new IOException("Failed to download agent JAR: " + agentReleaseAsset.getDownloadUrl(), e);
+        }
+    }
+
+    private @Nullable Path getJavaAgentPath() {
+        var agentDir = getOrCreateAgentDir();
+        return agentDir == null ? null : agentDir.resolve(AGENT_LINK_FILENAME);
+    }
+
+    /**
+     * @return The path of the directory, where the AppMap Java agent should be stored (~/.appmap/lib/java).
+     */
+    @Nullable Path getOrCreateAgentDir() {
+        var agentDirPath = Paths.get(System.getProperty("user.dir")).resolve(Paths.get("appmap", "lib", "java"));
```

Comment:
This should be `.appmap` instead of `appmap`.

Reply:
Thanks! This is fixed by  https://github.com/getappmap/appmap-intellij-plugin/pull/300/commits/0ead3e9c961bdf0a97243753352e42839d334b04